src/main/java/seedu/duke/models/schema/Schedule.java

```diff
@@ -286,6 +287,109 @@ public void deleteModule(String module) throws FailPrereqException, MissingModul
         //
     }
 
+    public void shiftModule(String module, int targetSem) throws IllegalArgumentException,
+            FailPrereqException, MandatoryPrereqException, MissingModuleException, InvalidObjectException {
+
+        if (targetSem < 1 || targetSem > MAXIMUM_SEMESTERS) {
+            throw new IllegalArgumentException("Please select an integer from 1 to 8 for semester selection");
+        }
+
+        int originalIndex = modulesPlanned.getIndexByString(module);
+
+        if (originalIndex == -1) {
+            throw new MissingModuleException("Module does not exist in schedule");
+        }
+
+        int originalSem = 1;
+        int moduleCount = modulesPerSem[0];
+
+        while ((moduleCount - 1) < originalIndex) {
+            moduleCount += modulesPerSem[originalSem];
+            originalSem += 1;
+        }
+
+        int indexToAdd = 0;
+        for (int i = 1; i < targetSem; i++) {
+            indexToAdd += this.modulesPerSem[i - 1];
+        }
+
+        // User input sem that module is already in
+        if (originalSem == targetSem) {
+            throw new IllegalArgumentException("Module is already in semester " + targetSem);
+        }
+
+        // If shifting module earlier
```

Comment:
i think your comments here make it very clear what you are trying to do which is very helpful for other developers :D